eden/scm/edenscm/git.py

```diff
@@ -153,6 +153,16 @@ def clone(ui, url, destpath=None, update=True, pullnames=None):
                     # If `git ls-remote --symref <url> HEAD` failed to yield a name,
                     # fall back to the using the names in the config.
                     pullnames = bookmod.selectivepullbookmarknames(repo)
+                default_publicheads = repo.ui.config('remotenames',
+                                     'publicheads').split(',') # ['remote/master', 'remote/main']
+                remote_publicheads = ['remote/' + path for path in pullnames]
+                all_publicheads = ','.join(sorted(set(default_publicheads + remote_publicheads)))
+
+                configfilename = repo.ui.identity.configrepofile()
+                rcutil.editconfig(repo.ui,
+                                  repo.localvfs.join(configfilename),
+                                  'remotenames', 'publicheads',
+                                  all_publicheads)
```

Comment:
what do you think?  I added a function `update_and_persist_repo_config` in `rcutil`, but maybe it would be better to add a method to `localrepository` so this line would become

```py
repo.update_and_persist_config(`remotenames`, `publicheads`, all_publicheads)
```

Reply:
I went ahead and did it.  Please see the latest version of this PR.

(If you want to see what I did before, check Version 4 in ReviewStack https://reviewstack.dev/facebook/sapling/pull/607)

Reply:
I prefer keep it in `git.py`, but I don't think it should be a blocker for this PR. I am importing it now.